checklist/src/main/scala/checklist/Message.scala

```diff
@@ -3,34 +3,29 @@ package checklist
 import cats.{Eq, Order}
 import cats.data.NonEmptyList
 
-sealed abstract class Message(val isError: Boolean, val isWarning: Boolean) {
-  def text: String
-  def path: Path
-
-  def prefix[A: PathPrefix](prefix: A): Message = this match {
-    case result @ ErrorMessage(_, path)   => result.copy(path = prefix :: path)
-    case result @ WarningMessage(_, path) => result.copy(path = prefix :: path)
-  }
-}
-
-final case class ErrorMessage(text: String, path: Path = PNil) extends Message(true, false)
+object Message {
```

Comment:
Main change in this file is that `ErrorMessage` and `WarningMessage` are no longer unified by `Message`

Reply:
Why not have a single `Message` type? The error-ness or warning-ness is now defined by the position in the type, so do they still need separate case classes?

Reply:
It could be anything that has an instance of something like a `Pathable` type class that provides the functionality to track error locations.